codes/quantum/oscillators/dual_rail.yml

```diff
@@ -0,0 +1,53 @@
+code_id: dual_rail
+
+physical: qubits
+logical: qubits
+
+name: 'Dual Rail'
+
+introduced: \cite{doi:10.1103/PhysRevA.52.3489}
+
+description: 'In a dual rail qubit encoding, single qubits are represented by Fock states
+in 2 modes. The logical \(\ket{0}\) state is represented by \(\ket{01}\),
+while the logical \(\ket{1}\) state is represented by \(\ket{10}\). The original
+paper that introduced this encoding seeks to implement Deutsch''s algorithm specifically,
+i.e., it does not provide a framework for universal quantum computation (although this can
+be realized using Kerr non-linearities along with beamsplitters), and it does not solve the
+general problem of error correction.
+In the original paper
+that introduced this encoding, error correction is performed by simply rejecting
+executions which result in error states (i.e, illegal results), and re-executing.'
+
+protection: 'This code cannot protect against general errors.
+It protects against amplitude damping due to a Caldeira-Leggett type coupling
+between the mode and the environment, with coupling constant \( \gamma \). When
+photon loss occurs, the output state does not agree with any of the possible correct
+outcomes, as it has a different photon number. Further, for Deutsch''s  problem specifically,
+this code protects against errors resulting in states that have the correct photon number, but
+in the wrong modes.'
+
+features:
+  general_gates:
+    - 'General gates are performed using beamsplitters and Kerr non-linearities.
+    Universal quantum computing can be achieved with photons in dual rail encodings using the
+    KLM protocol with only linear optical elements and photon detectors.'
+
+notes:
+  - 'For Deutsch''s problem with optical qubits, error correction using photon number detection
```

Comment:
- For first note, add ref to Deutsch's problem.
- Move second note to cousin with code_id: chuang-leung-yamamoto
- Merge last note into protection section; I think it is basically equivalent to saying this is an error detecting code against loss.

Reply:
fixed